datastore/mongo/application.go

```diff
@@ -154,24 +138,24 @@ func (db *appRepo) SearchApplicationsByGroupId(ctx context.Context, groupId stri
 	}
 
 	filter := bson.M{
-		"group_id":        groupId,
-		"document_status": datastore.ActiveDocumentStatus,
```

Comment:
Was there a reason we are filtering out deleted groups any more?

Reply:
@jirevwe I think it should be implicit. I made it the responsibility of the store. But there are several inconsistencies, some store methods add this scope others rely on the Caller to do set it. To achieve consistency, I think we should do it in another refactor PR.